### PR TITLE
Add recruitment banner on specific mainstream browse pages

### PR DESCRIPTION
**Pull request by hannako**
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

This PR is entirely based on the excellent work of @chao-xian https://github.com/alphagov/collections/pull/2630

Final commit wip.

https://trello.com/c/qPhlRiUx/779-launch-and-monitor-the-recruitment-banner-for-money-tax

### CODE REVIEW

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -26,6 +26,7 @@ $govuk-new-link-styles: true;
 @import 'govuk_publishing_components/components/image-card';
 @import 'govuk_publishing_components/components/input';
 @import 'govuk_publishing_components/components/inset-text';
+@import 'govuk_publishing_components/components/intervention';
```
Comment (andysellick) [inline app/assets/stylesheets/application.scss:29]: The intervention component also has a JavaScript file, I'd recommend including that as well in this commit.

File: app/views/second_level_browse_page/_links.html.erb [hunk 2]
```diff
@@ -1,6 +1,14 @@
 <div class="browse__inner <%= page.lists.curated? ? 'browse__inner--curated' : 'browse__inner--alphabetical' %>">
   <h2 tabindex="-1" class="browse__heading js-heading"><%= t("shared.browse.prefix") %> <%= page.title %></h2>
 
+  <%= render "govuk_publishing_components/components/intervention", {
+    suggestion_text: "Help improve GOV.UK",
+    suggestion_link_text: "Take part in user research",
```
Comment (andysellick) [inline app/views/second_level_browse_page/_links.html.erb:6]: I'm guessing there aren't translations yet for the `suggestion_text` or `suggestion_link_text` but could this text come from a translations file please?

File: app/assets/javascripts/browse-columns.js [hunk 3]
```diff
@@ -190,6 +191,7 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
     this.updateBreadcrumbs(state)
 
     this.changeColumnVisibility(3)
+    this.showBanner(state.slug)
```
Comment (andysellick) [inline app/assets/javascripts/browse-columns.js:194]: This call is in the `showSubsection` function but not the `showSection` function. I don't have a complete picture yet of the intended functionality here but just wanted to check that this was only needed here.

File: app/views/second_level_browse_page/_links.html.erb [hunk 4]
```diff
@@ -1,6 +1,14 @@
...
+    suggestion_link_url: "https://gdsuserresearch.optimalworkshop.com/treejack/35gap0ij",
```
Comment (hannako) [inline app/views/second_level_browse_page/_links.html.erb:7]: This needs to be updated to https://GDSUserResearch.optimalworkshop.com/treejack/lb5eu75l

File: app/assets/javascripts/browse-columns.js [hunk 5]
```diff
@@ -303,5 +305,16 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
     }
   }
 
+  BrowseColumns.prototype.showBanner = function (slug) {
```
Comment (andysellick) [inline app/assets/javascripts/browse-columns.js:308]: I'm confused why this needs JavaScript at all, but perhaps I've missed something so happy to discuss.

The logic to show/hide the intervention component is already present in the backend, and the JS gets its data from that same place (via JSON in an AJAX call, which includes a chunk of markup for the column to be rendered, which includes/doesn't include the intervention component as appropriate).

As an experiment, I checked out this branch locally, removed this JS and modified `_links.html.erb` as follows:

```
  <% if !@hide_recruitment_banner %>
    <%= render "govuk_publishing_components/components/intervention", {
      suggestion_text: "Help improve GOV.UK",
      suggestion_link_text: "Take part in user research",
      suggestion_link_url: "https://gdsuserresearch.optimalworkshop.com/treejack/35gap0ij",
      new_tab: true,
    } %>
  <% end %>
```

...and it worked as expected, showing the intervention banner only where it was required (and persisting through the JS cache that's part of the browse JS also as required).

(might need a bit of refactoring in this PR to change `if !@hide_recruitment_banner` to something clearer like `if @show_recruitment_banner`)

File: app/assets/javascripts/browse-columns.js (hunk 3 above)
Comment (chao-xian) [reply app/assets/javascripts/browse-columns.js]: Removed the JS - thanks Andy

File: app/assets/javascripts/browse-columns.js (hunk 5 above)
Comment (chao-xian) [reply app/assets/javascripts/browse-columns.js]: Yup trying this out now. Will refactor the name in a bit.

File: app/assets/stylesheets/application.scss (hunk 1 above)
Comment (chao-xian) [reply app/assets/stylesheets/application.scss:29]: Done

File: app/views/second_level_browse_page/_links.html.erb (hunk 2 above)
Comment (chao-xian) [reply app/views/second_level_browse_page/_links.html.erb]: I'd normally agree to do this, but we then have to add keys with missing translations to _all_ of the non-en files as [this test expects](https://github.com/alphagov/collections/blob/main/spec/lib/locales_validation_spec.rb). As the banner is temporary, can we get away with not?

File: app/assets/javascripts/browse-columns.js (hunk 5 above)
Comment (chao-xian) [reply app/assets/javascripts/browse-columns.js]: Done

File: app/views/second_level_browse_page/_links.html.erb (hunk 2 above)
Comment (andysellick) [reply app/views/second_level_browse_page/_links.html.erb]: Ah, yeah that's a bit of a pain, I hadn't realised that. In that case yes, that's fine.

Comment (chao-xian) [reply app/views/second_level_browse_page/_links.html.erb]: Thank you!

**Comment by chao-xian** [comment]
Works on integration and locally - something weird with Heroku

**Review by andysellick — APPROVED** [review]
We can't reproduce the error I found anywhere except in the Heroku preview, so I'm going to approve this for now to avoid blocking.

**Pull request merged**